app/routes/preprints/provider/preprint-detail.js

```diff
@@ -45,6 +46,19 @@ export default Route.extend({
                 return this.intermediateTransitionTo('page-not-found');
             }
         },
+        willTransition(transition) {
+            if (this.controller.get('userHasEnteredReview')) {
```

Comment:
Could this reuse `this.isPageDirty()`? It could someday get more complex, it would be nice to have it in a single place.